File: backend/src/main/java/equipo1/model/Especialidad.java
package equipo1.model;

import javax.persistence.DiscriminatorColumn;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;

@Entity(name="especialidad")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name="nombre")

public class Especialidad {
	@Id
	protected String nombre;
	protected int duracionCita;
	
	public String getNombre() {
		return nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	public int getDuracionCita() {
		return duracionCita;
	}
	public void setDuracionCita(int duracionCita) {
		this.duracionCita = duracionCita;
	}
	
}

File: frontend/cliente/src/app/entity/Especialidad.ts
export class Especialidad{
    _id?: string;
    nombre: string;
    duracionCita: number;
    listaMedicos: String[];

    constructor(){
        this._id="";
        this.nombre= "";
        this.duracionCita=null;
        this.listaMedicos=null;

    }
}